zone/client_packet.cpp

```diff
@@ -14765,3 +14773,177 @@ void Client::Handle_OP_ResetAA(const EQApplicationPacket *app)
 	}
 	return;
 }
+
+void Client::CheatDetected(CheatTypes CheatType, float x, float y, float z)
+{
+	switch (CheatType)
+	{
+	case MQWarp:
+		if (RuleB(Zone, EnableMQWarpDetector) && ((this->Admin() < RuleI(Zone, MQWarpExemptStatus) || (RuleI(Zone, MQWarpExemptStatus)) == -1)))
+		{
+			Message(13, "Large warp detected.");
+			char hString[250];
+			sprintf(hString, "/MQWarp with location %.2f, %.2f, %.2f", GetX(), GetY(), GetZ());
+			database.SetMQDetectionFlag(this->account_name, this->name, hString, zone->GetShortName());
```

Comment:
This needs to log to its own log category, I would create a MQDetection logging category and log there as well as the database table